src/main/webapp/script.js

```diff
@@ -1,8 +1,88 @@
- function fetch_auth_info(){
-    fetch('/AuthServlet').then(response => response.json()).then((logUrl) => {
-    console.log(logUrl);
-    document.getElementById("loglink").href = logUrl[1];
-    document.getElementById("loglink").innerText = logUrl[0];
-     });
- }
+var currentListings = []
+
+class Listing {
+  constructor(subject, desc, time) {
+    this.subject = subject;
+    this.description = desc;
+    this.time = time;
+  }
+}
+
+function loadListings() {
+  fetch_auth_info();
+  getListings();
+}
+
+function fetch_auth_info(){
+  fetch('/AuthServlet').then(response => response.json()).then((logUrl) => {
+  console.log(logUrl);
+  document.getElementById("loglink").href = logUrl[1];
+  document.getElementById("loglink").innerText = logUrl[0];
+    });
+}
+
+/**
+  * Fetches comments from datastore, converts them to JSON and then does 2 things:
+  * 1) Creates new Listing objects from them and saves them in a global variable
+  * 2) updates the DOM to add HTML representing each Listing object.
+  */
+function getListings() {
+  let container = document.getElementById('listings-container');
+
+  fetch('/listings').then(res => res.json())
+  .then((fetchedListings) => {
+    currentListings = [];
+
+    for (var l of fetchedListings) {
+      let newListing = Object.assign(new Listing(), l);
```

Comment:
Something I didn't know before: turned out you can access json fields with dot syntax jsonObject.field not just with indexing syntax jsonObject["field"]. This makes creating separate class (Listing) redundant.

Reply:
So would it be better to not have a class for it? What if these objects need some extended functionality like getting the date in multiple formats. With a class this could be easily achieved with class functions and very little modification of existing code. I guess my question is: It seems like for the future a class could be useful, but should I take it out for now because it's not needed yet?

Reply:
If you anticipate that the class will be useful, then keep it in the code. My point is that you don't have to create a class if all you need is access to fields.

Reply:
Got it. For now I don’t have any outstanding thoughts about how I could extend it so I’ll take it out.